packages/framework/presence/src/presence.ts

```diff
@@ -79,17 +79,31 @@ export interface ISessionClient<
 	 * @remarks
 	 * Connection id will change on reconnect.
 	 *
-	 * If {@link ISessionClient.getStatus} is {@link (SessionClientStatus:variable).Disconnected}, this will represent the last known connection id.
+	 * If {@link ISessionClient.getConnectionStatus} is {@link (SessionClientStatus:variable).Disconnected}, this will represent the last known connection id.
 	 */
-	connectionId(): ClientConnectionId;
+	getConnectionId(): ClientConnectionId;
 
 	/**
-	 * Get status of session client.
+	 * Get connection status of session client.
 	 *
-	 * @returns Status of session client.
+	 * @returns Connection status of session client.
 	 *
 	 */
-	getStatus(): SessionClientStatus;
+	getConnectionStatus(): SessionClientStatus;
+
+	/**
+	 * Set current client connection id.
+	 *
+	 * @param connectionId - New connection id.
+	 */
+	setConnectionId(connectionId: ClientConnectionId): void;
+
+	/**
+	 * Set connection status of session client.
+	 *
+	 * @param connected - True if client is connected.
+	 */
+	setConnectionStatus(connected: boolean): void;
```

Comment:
These are read only properties - the external interface should not have any set* methods.